### PR TITLE
[fix] fix a bug which will cause compiling error and json parse error when ifdef NGX_RTMP_POOL_DEBUG.

### PR DESCRIPTION
**Pull request by plainheart**
I've not been paying attention to the case when **NGX_RTMP_POOL_DEBUG** is defined for a long time. Until now, I have realized it will cause some serious compiling and parsing problem and fixed them.
After this fix, the JSON struct performed some little changes.

- `http-flv.servers` -> `http-flv.server`
- `http-flv.server` -> 
```js
{
    pool: {}, // ifdef NGX_RTMP_POOL_DEBUG 
    applications: []
}
```
- `http-flv.server.applications[i].live.streams[i].clients[i]` added a new key-value named `pool` (ifdef **NGX_RTMP_POOL_DEBUG**)

image

### CODE REVIEW

File: ngx_rtmp_stat_module.c [hunk 1]
```diff
@@ -343,7 +344,8 @@ ngx_rtmp_stat_get_pool_size(ngx_pool_t *pool, ngx_uint_t *nlarge,
 
 
 static void
-ngx_rtmp_stat_dump_pool(ngx_http_request_t *r, ngx_chain_t ***lll,
+ngx_rtmp_stat_dump_pool(ngx_rtmp_stat_loc_conf_t *slcf,
```
Comment (winshining) [inline ngx_rtmp_stat_module.c]: `slcf` pointer can be got via `r` and it's easy to get it, so no need to add a new variable.

Comment (plainheart) [reply ngx_rtmp_stat_module.c]: Sorry for my fault. Fixed.

File: ngx_rtmp_stat_module.c [hunk 2]
```diff
@@ -1046,14 +1053,19 @@ ngx_rtmp_stat_server(ngx_http_request_t *r, ngx_chain_t ***lll,
...
 
 #ifdef NGX_RTMP_POOL_DEBUG
     ngx_rtmp_stat_dump_pool(r, lll, cscf->pool);
+    if (slcf->format & NGX_RTMP_STAT_FORMAT_JSON) {
+        NGX_RTMP_STAT_L(",");
+    }
 #endif
 
+    if (slcf->format & NGX_RTMP_STAT_FORMAT_JSON) {
+        NGX_RTMP_STAT_L("\"applicatons\":[");
```
Comment (HeyJupiter) [inline ngx_rtmp_stat_module.c]: @winshining Hey, I found a typo here.

**Comment by winshining** [comment]
@HeyJupiter THX!

**Pull request merged**